content/lessons/chapter-8/building-blocks-4.tsx

```diff
@@ -1,86 +1,109 @@
 'use client'
 
-import { useProceed, useTranslations } from 'hooks'
-import { ChapterIntro, CodeExample, HolocatQuestion } from 'ui'
-
-import { Button } from 'shared'
-import { useEffect, useState } from 'react'
-import { getData } from 'api/data'
-import { Data } from 'types'
-import { chapters } from 'content/chapters'
+import { ScriptingChallenge, LessonInfo, CodeExample } from 'ui'
+import { EditorConfig } from 'types'
+import { useTranslations } from 'hooks'
+import { Text } from 'ui'
+import { useState } from 'react'
+import { getLessonKey } from 'lib/progress'
+import { useDataContext } from 'contexts/DataContext'
+import { getLanguageString } from 'lib/SavedCode'
+import { useAtom } from 'jotai'
+import { accountAtom } from 'state/state'
 
 export const metadata = {
-  title: 'chapter_four.address_one.title',
+  title: 'chapter_eight.building_blocks_four.nav_title',
+  navigation_title: 'chapter_eight.building_blocks_four.nav_title',
   key: 'CH8BBK4',
 }
 
 export default function BuildingBlocks4({ lang }) {
-  const proceed = useProceed()
   const t = useTranslations(lang)
+  const [account] = useAtom(accountAtom)
+  const { currentLanguage } = useDataContext()
+  const [privateKey, setPrivateKey] = useState('')
 
-  const [prevData, setPrevData] = useState<Data>({ lesson_id: '', data: '' })
-  const dataObject = prevData?.data ? prevData?.data : ''
-  const [isLoading, setIsLoading] = useState(true)
-  const [tooltipVisible, setTooltipVisible] = useState(false)
+  if (account && !privateKey) {
+    setPrivateKey(account?.private_key.toString())
+  }
 
-  const handleMouseEnter = () => {
-    setTooltipVisible(true)
+  const javascript = {
+    program: `
+console.log(privateKeyToPublicKey(\`${privateKey}\`))
+console.log("KILL")`,
+    defaultFunction: {
+      name: 'privateKeyToPublicKey',
+      args: ['privateKey'],
+    },
+    defaultCode: `const CODE_CHALLENGE_2_HEIGHT = 6929996;
+`,
+    validate: async (answer: string) => {
+      return [true, 'Nicely Done ']
+    },
   }
 
-  const handleMouseLeave = () => {
-    setTooltipVisible(false)
+  const python = {
+    program: `
+print(privatekey_to_publickey("${privateKey}"))
+print("KILL")`,
+    defaultFunction: {
+      name: 'privatekey_to_publickey',
+      args: [''],
+    },
+    defaultCode: `CODE_CHALLENGE_2_HEIGHT = 6929996
+hashes = Bitcoin.rpc("getblocksbyheight", CODE_CHALLENGE_2_HEIGHT)
+answer = None
+tx_count = float("inf")
+for bhash in hashes:
+    block = Bitcoin.rpc("getblock", bhash)
+    num = len(block["txs"])
+    if num < tx_count:
+        tx_count = num
+        answer = bhash
+print(answer)
```

Comment:
I want to try and rework how the function is a little since this requires the user to print to the terminal for this challenge, I'd rather make it a return function